tests/lib/api.js

```diff
@@ -37,4 +42,27 @@ describe("api", () => {
     it("should have SourceCode exposed", () => {
         assert.isFunction(api.SourceCode);
     });
+
+    describe("loadESLint", () => {
+        it("should be a function", () => {
+            assert.isFunction(api.loadESLint);
+        });
+
+        it("should return a Promise", () => {
+            assert.instanceOf(api.loadESLint(), Promise);
+        });
+
+        it("should return FlatESLint when useFlatConfig is true", async () => {
+            assert.strictEqual(await api.loadESLint({ useFlatConfig: true }), FlatESLint);
+        });
+
+        it("should return ESLint when useFlatConfig is false", async () => {
+            assert.strictEqual(await api.loadESLint({ useFlatConfig: false }), api.ESLint);
+        });
+
+        it("should return FlatESLint when useFlatConfig is not provided because we have eslint.config.js", async () => {
+            assert.strictEqual(await api.loadESLint(), FlatESLint);
+        });
```

Comment:
Can we create a test case where `api.loadESLint()` returns legacy eslint?

Reply:
Hmm, good question. Let me look into that.